### PR TITLE
Kzlprd 474 etq utilisateur jvp pouvoir avoir une liste de valeurs possibles pour une metadonnees de type string

### PR DESCRIPTION
**Pull request by cyrng**
## What does this PR do ?

Update metadata details structure to include new parameters :
- list of values
- type of list
- allow custom value

PR related to :
https://github.com/kuzzleio/iot-platform/pull/251
https://github.com/kuzzleio/iot-platform/pull/283

### CODE REVIEW

**Review by QuentinRousselet — APPROVED** [review]

File: doc/2/controllers/models/update-asset/index.md [hunk 1]
```diff
@@ -48,7 +48,12 @@ Method: PUT
                 description: string;
               };
             };
-            readOnly?: boolean;
+            definition?: {
+              isReadOnly?: boolean;
```
Comment (rolljee) [inline doc/2/controllers/models/update-asset/index.md:52]: I would rather call it readonly: boolean for simplicity

File: doc/2/controllers/models/update-asset/index.md [hunk 2]
```diff
@@ -48,7 +48,12 @@ Method: PUT
...
+              type?: string;
+              values?: string[] | number[] | boolean [];
+              isAllowedToCustom?: boolean;
```
Comment (rolljee) [inline doc/2/controllers/models/update-asset/index.md:55]: I would rather call it customValueAllowed: boolean

File: doc/2/controllers/models/update-asset/index.md (hunk 1 above)
Comment (cyrng) [reply doc/2/controllers/models/update-asset/index.md:52]: I'm ok with the changing but putting 'is', 'has', etc... for boolean is not a convention ?

File: doc/2/controllers/models/update-asset/index.md (hunk 2 above)
Comment (cyrng) [reply doc/2/controllers/models/update-asset/index.md:55]: Same answer as above

File: doc/2/controllers/models/update-asset/index.md (hunk 1 above)
Comment (etrousset) [reply doc/2/controllers/models/update-asset/index.md:52]: I agree with Ricky, the `is` isn't necessary, `readOnly` is enough

File: doc/2/controllers/models/write-asset/index.md [hunk 3]
```diff
@@ -49,7 +49,12 @@ Method: POST
                 description: string;
               };
             };
-            readOnly?: boolean;
+            definition?: {
+              isReadOnly?: boolean;
+              type?: string;
+              values?: string[] | number[] | boolean [];
+              isAllowedToCustom?: boolean;
```
Comment (etrousset) [inline doc/2/controllers/models/write-asset/index.md:56]:
```suggestion
              allowCustomValue?: boolean;
```

**Review by etrousset — CHANGES_REQUESTED** [review]

**Review by etrousset — APPROVED** [review]

**Pull request merged**